std/typecons.d

```diff
@@ -6896,3 +6896,190 @@ unittest
         ubyte, ubyte, T3, T3,
     );
 }
+
+/++
+ + Type constructor for head-const variables.
+ +
+ + Head-const variables cannot be directly mutated or rebound, but references
+ + reached through the variable are typed with their original mutability.
+ + It is equivalent to $(D final) variables in D1 and Java, as well as
+ + $(D readonly) variables in C#.
+ +
+ + When $(D T) is a $(D const) or $(D immutable) type, $(D HeadConst) aliases
+ + to $(D T).
+ +/
+struct HeadConst(T)
+    if (!is(T == const) && !is(T == immutable))
+{
+    private T HeadConst_value;
+    mixin Proxy!HeadConst_value;
```

Comment:
Thanks, I added the relevant test. Further attempts to break it are much appreciated!